### ISSUE TITLE
Is coreutils argument flag portability in scope? (macOS sed -i/uniq -D)

### ISSUE THREAD

**Issue by ChillerDragon** [koalaman/shellcheck#2902]
#### For bugs
- My shellcheck version (`shellcheck --version` or "online"):  0.9.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

Related issues:
https://github.com/koalaman/shellcheck/issues/973
https://github.com/koalaman/shellcheck/issues/479

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

# works on my linux breaks on my mac

sed -i "s/foo/bar/" file.txt
uniq -D file.txt

```

#### Here's what I wanted or expected to see:

Flag X is not portable (might break on BSD/macOS)

**Comment by Ordoviz** [comment]
I want this, too! You probably would need a way to tell shellcheck which userspace(s) (e.g. GNU, BSD, busybox, macOS, POSIX) you are targeting, though: https://github.com/koalaman/shellcheck/issues/1455#issuecomment-453875211

**Comment by ChillerDragon** [comment]
Also interesting on macOS sed can not handle invalid utf8. That includes the nullbytes which are commonly used as separators to allow for filenames with newlines.

```
find . -print0 | xargs -0 sed 's/a/b/g' > /dev/null
```

Shows the following error on macOS

```
sed: RE error: illegal byte sequence
```

https://stackoverflow.com/questions/21470124/invalid-command-code-error-from-sed-after-running-find-and-sed-on-mavericks#21470489
https://stackoverflow.com/questions/19242275/re-error-illegal-byte-sequence-on-mac-os-x

**Comment by barbarkaragul-oss** [comment]
Hi. I am not a ShellCheck developer, just a user who got bitten by this many times.

I made a small side project that may help with the "who maintains the list" problem from #479. Instead of writing the list by hand, a GitHub Actions job runs every flag of 81 common tools (and 86 find primaries) on ubuntu-latest, macos-latest and alpine:3.20 once a week. If the tool answers with its own "unknown option" message, the flag is recorded as missing on that platform. The output is one JSON file, MIT licensed:

https://github.com/barbarkaragul-oss/will-it-run-on-a-mac (data/flags.json)

There is also a web page where you paste a script and see which flags break on macOS or BusyBox, but for this issue the useful part is probably the data. It is not a ShellCheck check, and it only knows if a flag exists, not what it means. If it helps with a future userland target option (GNU / BSD / BusyBox, like suggested above), feel free to take anything from it.

One thing I learned from the runs: macOS 26 now has `sort -V`, `uniq -D` and `readlink -f`, but still no `sed -z`, `grep -P`, `date -d` or `find -printf`.

If this is off topic here, sorry, please ignore.
